src/Client.ts

```diff
@@ -279,7 +297,22 @@ export default class Client {
   }
 
   private async sendWakuMessage(msg: WakuMessage): Promise<void> {
-    const ack = await this.waku.lightPush.push(msg)
+    // Waku randomly selects a peer from the Peerstore to send the message to. To ensure this is the
+    // same peer to which we authenticated to, a random peer is selected at this context and then
+    // passed in to LightPush to ensure a match
+    const dstPeer = await this.waku.lightPush.randomPeer
```

Comment:
Hm, is this going to make clients use all lightpush nodes? Wouldn't we want to have the clients stick with fewer rather than more?

Reply:
I agree. I think not randomly selecting a peer would be preferred, but that seems like a bigger change that I'd want to handle outside of this PR. Waku randomly selects the peer by default, so this intends to matches that behavior. 

I also think in practice this is not a big deal as it seems. I traced the SDK looking for where peers get added to the `libp2p.peerStore` (which is where randomPeer samples from) and couldn't find any references in`xmtp-js` and `js-waku`. 

I've yet to confirm but it appears the only place place a peer gets added to the `peerStore.addressBook` is in `Libp2p.dialer.dial`. This would mean that its only randomly selecting peers to which its previously connected to. 

ref: https://github.com/libp2p/js-libp2p/blob/e6f646ed361d231ec2306fee40fcb87c18af0c41/src/connection-manager/dialer/index.ts#L171

Reply:
Yes. In practice it is always one node per client. That node is picked randomly from the bootstrap list, but then stays consistent for the browser session. Maybe Waku is going to fix this at some point, but that's the way it works right now.

Reply:
Sounds good 👍